MIGRATION_GUIDE.md

```diff
@@ -1,9 +1,170 @@
 # Migration guide
 
-This document is meant to help you migrate your Terraform config to new newest version. In migration guides we will only 
-describe deprecations or breaking changes and help you to change your configuration to keep the same (or similar) behaviour
+This document is meant to help you migrate your Terraform config to the new newest version. In migration guides, we will only 
+describe deprecations or breaking changes and help you to change your configuration to keep the same (or similar) behavior
 across different versions.
 
+## vX.XX.X -> v0.85.0
+
+### Migration from old grant resources to new ones
+
+In recent changes, we introduced new grant resources intended to replace old grant solutions. Here's some of the useful
+information that may help during the migration of the grant resources. Here's an example of the `snowflake_database_grant` to `snowflake_grant_privileges_to_account_role` migration.
+The migration can be done in two ways. Either you can remove old grant resources and replace them with new ones or perform
+more complicated migration, but without revoking any grant (no downtime migration). We'll focus on the second one as the first approach
+is pretty straight forward. As an example we'll take `snowflake_database_grant` that grants one privilege to two roles:
+
+```terraform
+resource "snowflake_database_grant" "old_resource" {
+  depends_on = [ snowflake_database.test, snowflake_role.a, snowflake_role.b ]
+  database_name = snowflake_database.test.name
+  privilege = "USAGE"
+  roles = [ snowflake_role.a.name, snowflake_role.b.name ]
+}
+```
+
+#### 1. terraform list
+
```

Comment:
i would probably put a few sentences why you are doing this. "first we need to list all the grant resources that will need to be migrated. We can do that by running the `terraform state list` command".

Tip: for larger configurations, its best to filter the results using the grep commands;